mobify/mobify-code-style
# Fix eslint for Atom

Status: **Ready for Review**
Owner: @marlowpayne
Reviewers: @donnielrt @Helen-Mobify 

## Changes
Atom currently has issues parsing our eslint settings due to improper quote marks in certain settings.
![atomeslintconfigerror](https://cloud.githubusercontent.com/assets/1747353/8483220/058eb862-20a4-11e5-910d-55432a3392c6.png)
In addition, Atom will not overwrite default eslint settings unless they are explicitly whitelisted in the supplied eslint config. These changes:
- Fix rules that Atom previously could not parse correctly
- Extend our eslint config on top of a blank slate of disabled default rules

These changes are compatible with our usage of eslint in Adaptive.js projects.

## Todos:

### Feedback:
_none so far_

## How to Test
A couple things to test here:
### Test on-the-fly linting in Atom
- Download and install [Atom](https://atom.io/)
- Install the [linter](https://atom.io/packages/linter) package
- Install the [linter-eslint](https://atom.io/packages/linter-eslint) package
- Update your local copy of the code-style by running `npm install -g git+ssh://git@github.com:mobify/mobify-code-style.git#fix-eslint-for-atom`
- Create a symlink to the updated eslint config files by navigating to your main projects directory (holding all your Mobify projects) and running:

`ln -s /path/to/your/global/node_modules/mobify-code-style/javascript/.eslintrc ./.eslintrc`
and
`ln -s /path/to/your/global/node_modules/mobify-code-style/javascript/.eslintrc-reset ./.eslintrc-reset`

(Default path is usually `/usr/local/lib/node_modules/`)

- Open your Atom settings for the `linter-eslint` package and make sure your path is set for the eslint rules
![rulespath](https://cloud.githubusercontent.com/assets/1747353/8483662/089455aa-20a7-11e5-88b8-e413260455ef.png)
- Restart Atom
- Open a JS file and check that linting works

### Test that project linting still works
- Open an Adaptive.js project built from the generator version 1.8 or higher (uses eslint)
- Update `package.json` to use this branch's version of eslint config by changing the entry for `mobify-code-style` to:

`"mobify-code-style": "git+ssh://git@github.com:mobify/mobify-code-style.git#fix-eslint-for-atom",`
- Refresh the project node modules with: `rm -rf node_modules && npm install`
- Verify linting runs against the new rules with: `grunt lint`
